plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/store-location.tsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import { SETTINGS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { getCountryCode } from '~/dashboard/utils';
+import { hasCompleteAddress, SettingsSelector } from '../../tax/utils';
+import { default as StoreLocationForm } from '~/tasks/fills/steps/location';
+
+export const StoreLocation: React.FC< {
+	nextStep: () => void;
+} > = ( { nextStep } ) => {
+	const { createNotice } = useDispatch( 'core/notices' );
+	const { updateAndPersistSettingsForGroup } =
+		useDispatch( SETTINGS_STORE_NAME );
+	const { generalSettings, isResolving } = useSelect( ( select ) => {
+		const { getSettings, hasFinishedResolution } = select(
+			SETTINGS_STORE_NAME
+		) as SettingsSelector;
+
+		return {
+			generalSettings: getSettings( 'general' )?.general,
+			isResolving: ! hasFinishedResolution( 'getSettings', [
+				'general',
+			] ),
+		};
+	} );
+
+	useEffect( () => {
+		if ( isResolving || ! hasCompleteAddress( generalSettings ) ) {
+			return;
+		}
+		nextStep();
+	}, [ generalSettings, isResolving, nextStep ] );
+
+	if ( isResolving ) {
+		return null;
+	}
+
+	return (
+		<StoreLocationForm
+			onComplete={ ( values: { [ key: string ]: string } ) => {
+				const country = getCountryCode( values.countryState );
+				recordEvent( 'tasklist_tax_set_location', {
```

Comment:
Do we still need to emit this **tax** event in the shipping task?

Reply:
Nope, thanks for catching it! I think keeping the track but renaming `tax` to `shipping_recommendation` would be good for now.